File: spring-reactive/spring-webflux-coroutine/blockhound-java/src/test/java/com/toy/blockhoundjava/controller/TestControllerTest.java
package com.toy.blockhoundjava.controller;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;
import reactor.blockhound.BlockHound;

import static org.junit.jupiter.api.Assertions.*;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@AutoConfigureWebTestClient
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestControllerTest {

  @Autowired
  private WebTestClient client;

  @BeforeAll
  void beforeAll() {
    BlockHound.install();
  }

  @DisplayName("call block api")
  @Test
  void block() {
    client.get()
      .uri("/test/block")
      .exchange()
      .expectStatus().is5xxServerError();
  }

  @DisplayName("call non-block api")
  @Test
  void nonBlock() {
    client.get()
      .uri("/test/non-block")
      .exchange()
      .expectStatus().isOk();
  }

  @DisplayName("call allow-block api")
  @Test
  void allowBlock() {
    client.get()
      .uri("/test/allow-block")
      .exchange()
      .expectStatus().isOk();
  }

}

File: spring-jacoco/src/test/kotlin/com/toy/springjacoco/controller/TestControllerTest.kt
package com.toy.springjacoco.controller

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.TestConstructor
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.get

@SpringBootTest
@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
@AutoConfigureMockMvc
class TestControllerTest(
  private val mockMvc: MockMvc
) {

  @Test
  fun test() {
    mockMvc.get("/test")
      .andExpect { status { isOk() } }
  }
}
